### PR TITLE
Admin Panel

### LINKED ISSUE #92: Buildout FE for Admin User Management Page
**Is your feature request related to a problem? Please describe.**

As an admin, I should be able to:

- See all the users in the platform
- Approve users
- Delete users
- Change role of existing users (upgrade or downgrade privilege)

**Describe the solution you'd like**

Since all the work has been done on the backend, what's left is purely buildout of the UI for each  of the above components.

### PR DESCRIPTION
**Pull request by JoshBurke**
## Status:

:rocket: Ready

## Description

Adds admin panel which shows list of users: pending users, and currently active users.
Refactors modal into more general modal component and special cases of user & resource modals.

Fixes #92 

## Todos

- [x] Generate table with user cards
- [x] User edit/approve modals
- [x] Refactor resource/user modals

### CODE REVIEW

File: frontend/src/components/Navbar/index.jsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const Navbar = props => {
       <div className="nav-links">
         <Link to="/">Map</Link>
         <Link to="/directory">Directory</Link>
+        <Link to="/users">Account Management</Link>
```
[the remark this answers is not in the document]
Comment (JoshBurke) [reply frontend/src/components/Navbar/index.jsx]: good idea - pushing changes

File: frontend/src/pages/AdminView/index.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { refreshAllUsers } from "../../utils/api";
+import { connect } from "react-redux";
+import { userSelector } from "../../redux/selectors/users";
+import UserCard from "./UserCard";
+import "./styles.scss";
+
+const UserManager = props => {
+  useEffect(() => {
+    refreshAllUsers();
+    console.log("hooook");
```
[the remark this answers is not in the document]
Comment (JoshBurke) [reply frontend/src/pages/AdminView/index.jsx]: lol ill get rid of this

File: frontend/src/redux/selectors/users.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const userSelector = state => state.users;
```
[the remark this answers is not in the document]
Comment (JoshBurke) [reply frontend/src/redux/selectors/users.js]: thanks for the link 👌👌👌

File: frontend/src/components/Modal/index.jsx [hunk 4]
```diff
@@ -1,192 +1,33 @@
...
+        <img id="close-image" src={Close} alt="close" />
+      </Button>
+    </ModalHeader>
+    <ModalBody
+      style={{
+        maxHeight: "calc(100vh - 210px)",
+        overflowY: "auto"
+      }}
+    >
+      {props.children}
```
Comment (josh-byster) [inline frontend/src/components/Modal/index.jsx]: 💯

File: frontend/src/components/Modal/ResourceModal/index.jsx [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+              <input
+                ref={register({ required: true })}
+                type="text"
+                name="companyName"
+                defaultValue={props.resource.companyName}
+                className="modal-input-field"
+                disabled={!props.editable}
+              />
+              {/* errors will return when field validation fails  */}
+              {errors.companyName && <span>This field is required</span>}
```
Comment (josh-byster) [inline frontend/src/components/Modal/ResourceModal/index.jsx]: Well done adding a validation step here!

**Comment by josh-byster** [comment]
Merging in latest Cypress tests to run as part of CI to verify functionality on this feature before merge.

**Pull request merged**